### PR TITLE
702 add user role in clan req

### PR DESCRIPTION
**Pull request by Rogerdin13**
added new ViewModel in backend for route response (/users/self/clans) and changed query to map from ClanMemeber to get clan and role in response.

changed frontend models to match new backend one and fixed default.vue accordingly.

### CODE REVIEW

File: src/WebUI/src/layouts/default.vue [hunk 1]
```diff
@@ -62,7 +62,7 @@ onBeforeUnmount(() => {
 });
 
 await Promise.all(promises);
-await userStore.getUserClanMember(); // TODO: get the clan role in the query `users/self/clans`
+await userStore.getUserClanMember();
```
Comment (or2e) [inline src/WebUI/src/layouts/default.vue:65]: the meaning of this task is to remove this method (delete, also from auto tests) that is.

earlier, to find out the role in the clan, we received a list of all clan members, ok?

https://github.com/verdie-g/crpg/blob/fc4a5b392d0e23e437b821037442af6cc4e6d3f5/src/WebUI/src/stores/user.ts#L73

Comment (Rogerdin13) [reply src/WebUI/src/layouts/default.vue:65]: ah, got it, will look into it again

File: src/WebApi/Controllers/UsersController.cs [hunk 2]
```diff
@@ -448,7 +448,7 @@ public Task<ActionResult<Result<UserItemViewModel>>> BuyItem([FromBody] BuyItemC
     /// Gets user clan or null.
     /// </summary>
     [HttpGet("self/clans")]
-    public Task<ActionResult<Result<ClanViewModel>>> GetUserClan()
+    public Task<ActionResult<Result<UserClanViewModel>>> GetUserClan()
```
Comment (verdie-g) [inline src/WebApi/Controllers/UsersController.cs:451]: It looks like it's the relevant change to the PR here. Could you please discard the rest.

**Review by verdie-g — CHANGES_REQUESTED** [review]
Nice job 👌

File: src/WebApi/Controllers/UsersController.cs (hunk 2 above)
Comment (Rogerdin13) [reply src/WebApi/Controllers/UsersController.cs:451]: Whole ClanViewModel and references u mean?

Comment (Rogerdin13) [reply src/WebApi/Controllers/UsersController.cs:451]: ah i overlooked the autoformatting stuff, sorry, added 6ca523d to revert layout changes

File: src/WebUI/src/services/users-service.ts [hunk 3]
```diff
@@ -8,12 +8,12 @@ import {
...
 } from '@/models/user';
 import Platform from '@/models/platform';
-import { type ClanEdition } from '@/models/clan';
+import { Clan, type ClanEdition, type ClanMemberRole } from '@/models/clan';
 import { type RestrictionWithActive } from '@/models/restriction';
 
 import { get, post, put, del } from '@/services/crpg-client';
 import { getActiveJoinRestriction, mapRestrictions } from '@/services/restriction-service';
-import { mapClanResponse } from '@/services/clan-service';
+import { mapClanResponse } from './clan-service';
```
Comment (or2e) [inline src/WebUI/src/services/users-service.ts:16]: return it to the way it was before ;)
Or we will have to write that in all services

File: src/WebUI/src/services/users-service.ts [hunk 4]
```diff
@@ -8,12 +8,12 @@ import {
   type UserItemRank,
 } from '@/models/user';
 import Platform from '@/models/platform';
-import { type ClanEdition } from '@/models/clan';
+import { Clan, type ClanEdition, type ClanMemberRole } from '@/models/clan';
```
Comment (or2e) [inline src/WebUI/src/services/users-service.ts:11]: Clan not used

**Review by or2e — APPROVED** [review]

File: src/Application/Clans/Queries/GetUserClanQuery.cs [hunk 5]
```diff
@@ -33,7 +33,7 @@ public async Task<Result<ClanViewModel>> Handle(GetUserClanQuery req, Cancellati
                 return new(CommonErrors.UserNotFound(req.UserId));
             }
 
-            return new(_mapper.Map<ClanViewModel>(user.ClanMembership?.Clan));
+            return new(_mapper.Map<UserClanViewModel>(user.ClanMembership));
```
Comment (verdie-g) [inline src/Application/Clans/Queries/GetUserClanQuery.cs:36]: Shouldn't you update the tests? I'm surprised it compiles with that change.

File: src/Application/Clans/Models/UserClanViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+using Crpg.Application.Common.Mappings;
+
+namespace Crpg.Application.Clans.Models;
+using Crpg.Application.Common.Mappings;
+using Crpg.Domain.Entities.Clans;
+
+
+public record UserClanViewModel : IMapFrom<ClanMember>
```
Comment (verdie-g) [inline src/Application/Clans/Models/UserClanViewModel.cs:8]: Can you reorder the statements
1. using
2. namespace
3. record

**Review by verdie-g — CHANGES_REQUESTED** [review]
Sorry overlooked a few things

File: src/Application/Clans/Models/UserClanViewModel.cs (hunk 6 above)
Comment (Rogerdin13) [reply src/Application/Clans/Models/UserClanViewModel.cs]: done, sorry, 1st time not having VS hold my hand when creating files, my bad :D

**Comment by or2e** [comment]
closed https://github.com/verdie-g/crpg/issues/702

**Pull request merged**